keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -242,15 +249,19 @@ open class SqlResourceRepository(
       if (event.ignoreRepeatedInHistory) {
         val previousEvent = txn
           .select(EVENT.JSON)
-          .from(EVENT, RESOURCE)
-          .where(EVENT.SCOPE.eq(Scope.RESOURCE.name))
-          .and(RESOURCE.ID.eq(event.id))
-          .and(RESOURCE.UID.eq(EVENT.UID))
+          .from(EVENT)
+          .where(EVENT_JSON_APPLICATION.eq(event.application))
+          // look for resource events that match the resource...
+          .and(EVENT.SCOPE.eq(Scope.RESOURCE.name)
+            .and(EVENT_JSON_RESOURCE_ID.eq(event.id))
+          )
+          // ...or application events that match the application as they apply to all resources
+          .or(EVENT.SCOPE.eq(Scope.APPLICATION.name))
```

Comment:
I need to add a condition here for the application event being later than the last resource event".

Reply:
Actually, this should work because of the descending sort order.